src/ripple/rpc/handlers/AccountTx.cpp

```diff
@@ -373,19 +373,20 @@ populateProtoResponse(
                             closeTime->time_since_epoch().count());
                     if (txnMeta)
                     {
-                        if (!txnMeta->hasDeliveredAmount())
+                        RPC::convert(*txnProto->mutable_meta(), txnMeta);
+
+                        auto amt = getDeliveredAmount(
+                            context,
+                            txn->getSTransaction(),
+                            *txnMeta,
+                            txn->getLedger());
+                        if (amt)
                         {
-                            std::optional<STAmount> amount = getDeliveredAmount(
-                                context,
-                                txn->getSTransaction(),
-                                *txnMeta,
-                                txn->getLedger());
-                            if (amount)
-                            {
-                                txnMeta->setDeliveredAmount(*amount);
-                            }
+                            RPC::convert(
+                                *txnProto->mutable_meta()
+                                     ->mutable_delivered_amount(),
+                                *amt);
```

Comment:
I'm a little confused. The old code looks correct to me. Is the problem that `RPC::convert` does not convert the `delivered_amount`? If so, why don't we fix `RPC::convert`? (If not, can you explain what's wrong with the old code?)

Reply:
The way I read this is that in the old code if `DeliveredAmount` was present, then `delivered_amount` was not inserted.  I think the intention is that `delivered_amount` is always provided so client code doesn't have to check two places.  No guarantee I got that right, though.

Reply:
`RPC::convert` does not convert `delivered_amount`. I thought about adding a conversion for `delivered_amount` inside `RPC::convert(org::xrpl::rpc::v1::Meta&,std::shared_ptr<TxMeta>)`, but the issue is that `delivered_amount` is not always present in the `TxMeta` object, but is sometimes populated from the `sfAmount` field of the transaction object. I didn't like the idea of having to remember to call `TxMeta::setDeliveredAmount()` before calling `RPC::convert()`. The code needs to call the `getDeliveredAmount()` helper function anyways, so I preferred setting `delivered_amount` directly, instead of altering the `TxMeta` object and passing it to `RPC::convert()`.
I don't feel strongly about this, so am willing to change the code if the other way is clearer.

Reply:
@scottschurr When I say "the old code looks correct to me" I mean I'd expect `RPC::convert` to handle `delivered_amount`. The fix here appears to compensate for a bug in `RPC::convert`. So unless there's a reason `RPC::convert` doesn't handle this, I'd keep this code as it was before and fix `RPC::convert`.

Reply:
@cjcobb23 It does seem weird that `RPC::convert` doesn't convert `delivered_amount` even if present. I vote that we change that.

Reply:
fixed in `d10a4a02`; `RPC::convert` converts `delivered_amount` if present.